### ISSUE TITLE
[Feature Request] Matrix compatible frontend

### ISSUE THREAD

**Issue by rodneyrod** [hanzz/spectrum2#182]
Since this is aspiring to be the 'replacement' for XMPP as well as a sort of open source Slack, it would make sense to support this as a frontend.

**Comment by go2null** [comment]
FYI - Libpurple protocol plugin for matrix  https://github.com/matrix-org/purple-matrix/

**Comment by rodneyrod** [comment]
I meant as a replacement for XMPP or Slack as a frontend, not as a plugin.

**Comment by vitalyster** [comment]
@rodneyrod do they have stabilised their federated api?

**Comment by rodneyrod** [comment]
@vitalyster From what I can see, they have. They're currently working on switching from one reference implementation to another, but they're only working on making both communicate with each other so it would seem the API is stable.

**Comment by vitalyster** [comment]
I don't think so. On the https://matrix.org/docs/spec/ page all specs except client-server are still marked unstable (and still have empty "work in progress" sections :))

**Comment by rodneyrod** [comment]
I wouldn't be the one to ask, I'm no dev I just want Spectrum to hook into this since it's going to be replacing XMPP.

**Comment by uhoreg** [comment]
It's probably better to use the Application Service (AS) API (analogous to XMPP components), rather than the Federation API.  The AS API seems to be pretty stable, and there are a number of bridges already using it.

**Comment by vitalyster** [comment]
@uhoreg spectrum have server and component modes and it will be good to implement both, also I'm interested in matrix server implementation more

**Comment by rodneyrod** [comment]
I think the APIs have developed some more since this ticket was opened.

**Comment by MurzNN** [comment]
Matrix.org started to build bridge using libpurple: https://github.com/matrix-org/matrix-appservice-purple
